source/views/menus/menu-github.tsx

```diff
@@ -1,113 +1,107 @@
-import * as React from 'react'
+import {useEffect, useState} from 'react'
 import {timezone} from '@frogpond/constants'
-import {NoticeView, LoadingView} from '@frogpond/notice'
+import {LoadingView, NoticeView} from '@frogpond/notice'
 import {FoodMenu} from '@frogpond/food-menu'
 import type {TopLevelViewPropsType} from '../types'
 import moment from 'moment-timezone'
-import type {Moment} from 'moment-timezone'
 import sample from 'lodash/sample'
-import fromPairs from 'lodash/fromPairs'
-import filter from 'lodash/filter'
 import type {
-	MenuItemType,
 	MasterCorIconMapType,
-	StationMenuType,
 	MenuItemContainerType,
+	MenuItemType,
 	ProcessedMealType,
+	StationMenuType,
 } from './types'
 import {upgradeMenuItem, upgradeStation} from './lib/process-menu-shorthands'
 import {API} from '@frogpond/api'
 import {fetch} from '@frogpond/fetch'
+import groupBy from 'lodash/groupBy'
 
 type Props = TopLevelViewPropsType & {
 	name: string
 	loadingMessage: string[]
 }
 
-type State = {
-	error?: Error
-	loading: boolean
-	now: Moment
-	foodItems: MenuItemContainerType
-	corIcons: MasterCorIconMapType
-	meals: ProcessedMealType[]
-}
-
-export class GitHubHostedMenu extends React.PureComponent<Props, State> {
-	state: State = {
-		error: undefined,
-		loading: true,
-		now: moment.tz(timezone()),
-		foodItems: {},
-		corIcons: {},
-		meals: [],
-	}
-
-	componentDidMount() {
-		this.fetchData()
-	}
+export function GitHubHostedMenu(props: Props): JSX.Element {
+	let [error, seterror] = useState<Error | null>(null)
+	let [loading, setloading] = useState(true)
+	let [now, setnow] = useState(moment.tz(timezone()))
+	let [foodItems, setfoodItems] = useState<MenuItemContainerType>({})
+	let [corIcons, setcorIcons] = useState<MasterCorIconMapType>({})
+	let [meals, setmeals] = useState<ProcessedMealType[]>([])
 
-	fetchData = async () => {
-		this.setState({loading: true})
+	useEffect(() => {
+		;(async () => {
+			setloading(true)
 
-		let container = await fetch(API('/food/named/menu/the-pause')).json<{
-			data: {
-				foodItems: MenuItemType[]
-				stationMenus: StationMenuType[]
-				corIcons: MasterCorIconMapType
+			let container
```

Comment:
Can we give this a type annotation?

Reply:
That's probably a better spot for it. Sure thing.

Reply:
Done!